odbc/src/api/data.rs

```diff
@@ -1254,15 +1261,60 @@ mod unit {
 
         #[test]
         fn positive_values_without_truncation_to_i32_succeed() {
-            let bson_double_to_i32 = Bson::Double(42.).to_u32();
-            let bson_int64_to_i32 = Bson::Int64(42).to_u32();
-            let bson_int32_to_i32 = Bson::Int32(42).to_u32();
+            let bson_double_to_i32 = Bson::Double(42.).to_i32();
```

Comment:
I've refactored the tests so that the main supported types regarding bounds checking are handled - strings, int64, and int32. Notably, doubles are missing, but strings are converted to doubles and handled that way.